processing/src/main/java/org/apache/druid/guice/ExtensionsConfig.java

```diff
@@ -20,35 +20,63 @@
 package org.apache.druid.guice;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Sets;
 
+import javax.annotation.Nullable;
 import javax.validation.constraints.NotNull;
+
 import java.util.LinkedHashSet;
+import java.util.List;
 
 /**
+ * Configuration for Druid extensions.
  */
 public class ExtensionsConfig
 {
   public static final String PROPERTY_BASE = "druid.extensions";
+  public static final String DEFAULT_EXTENSIONS_DIR = "extensions";
 
   @JsonProperty
   @NotNull
   private boolean searchCurrentClassloader = true;
 
+  /**
+   * The location of the "primary" extensions directory. If the path is relative
+   * (as in the default), then the path is relative to the Druid directory
+   * (assuming the Druid product directory is indicated by the working directory.)
+   * <p>
+   * The value can be blank to indicate to ignore this value and only use the path.
+   * (If the value is omitted from the config file, it defaults to "extensions", so
+   * we need a blank value to say to ignore this property.)
+   */
+  @JsonProperty
+  private String directory = DEFAULT_EXTENSIONS_DIR;
+
+  /**
+   * Extensions path. Items may be relative to the Druid home directory, or absolute.
+   * Extensions are resolved in order along the path: the first match wins. If
+   * {@link #directory} is set, it becomes the first element on the full path.
+   * A typical use is to let {@code directory} point to {@code $DRUID_HOME/extensions},
+   * and use {@link path} to point to extra extensions mounted into a Docker container.
+   * Primarily for testing.
+   */
   @JsonProperty
-  private String directory = "extensions";
+  @Nullable
+  private List<String> path;
```

Comment:
Please see my related comment on calling this `paths` or `pathList`. It's not apparent to me without looking at its type

Reply:
I see the point. We generally have documentation and examples since our property names don't include details. For example, `fooTimeout` often doesn't say `fooTimeoutMs`, `fooSize` doesn't say `fooSizeBytes` and `loadList` doesn't say `loadArrayOfString`.

That said, I'm open to suggestions if we feel folks can't look up the field in docs or in an example.